packages/gatsby-link/index.d.ts

```diff
@@ -1,6 +1,17 @@
 import * as React from "react"
 import { NavigateFn, LinkProps } from "@reach/router" // These come from `@types/reach__router`
 
+declare global {
```

Comment:
The types should be autogenerated from the source now and the manual file should no longer exist